tycho-client/src/snapshot.rs

```diff
@@ -0,0 +1,96 @@
+//! Snapshot request and response types for retrieving protocol state and contract storage.
+//!
+//! This module contains higher-level abstractions used by the client library to fetch
+//! complete snapshots of protocol state. These are not DTOs - they're constructed and
+//! consumed entirely in Rust code.
+
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+use tycho_common::{
+    dto::{
+        Chain, EntryPointWithTracingParams, ResponseAccount, ResponseProtocolState, TracingResult,
+    },
+    Bytes,
+};
+
+/// Request body for fetching a snapshot of protocol states and VM storage.
+///
+/// This is a higher-level struct used by the client library to coordinate fetching
+/// multiple pieces of related data (protocol states, contract storage, TVL, entry points).
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct SnapshotRequestBody {
+    #[serde(default)]
+    pub chain: Chain,
+    /// Protocol system name, required for correct state resolution
+    #[serde(alias = "protocolSystem")]
+    pub protocol_system: String,
+    /// Component IDs to fetch protocol states for
+    #[serde(alias = "componentIds", default)]
+    pub component_ids: Vec<String>,
+    /// Contract addresses to fetch VM storage for
+    #[serde(alias = "contractIds", default)]
+    pub contract_ids: Vec<Bytes>,
+    /// Block number for versioning
+    #[serde(alias = "blockNumber")]
+    pub block_number: u64,
+    /// Whether to include balance information
+    #[serde(alias = "includeBalances", default)]
+    pub include_balances: bool,
+    /// Whether to fetch TVL data
+    #[serde(alias = "includeTvl", default)]
+    pub include_tvl: bool,
+}
+
+impl SnapshotRequestBody {
+    pub fn new(
+        chain: Chain,
+        protocol_system: String,
+        component_ids: Vec<String>,
+        contract_ids: Vec<Bytes>,
+        block_number: u64,
+        include_balances: bool,
+        include_tvl: bool,
```

Comment:
Can we make these true by default and add methods that set them to false. So usage would look like this:

```rust
client.get_snapshot(
    SnapshotParams::new(
        "uniswap-v4",
        ...
    )
    .include_tvl(false)
    .include_balances(false)
)
```